src/java.base/share/classes/jdk/internal/foreign/abi/aarch64/TypeClass.java

```diff
@@ -58,15 +61,38 @@ static boolean isRegisterAggregate(MemoryLayout type) {
         return type.bitSize() <= MAX_AGGREGATE_REGS_SIZE * 64;
     }
 
+    static List<MemoryLayout> scalarLayouts(GroupLayout gl) {
+        List<MemoryLayout> out = new ArrayList<>();
+        scalarLayoutsR(out, gl);
+        return out;
+    }
+
+    private static void scalarLayoutsR(List<MemoryLayout> out, GroupLayout gl) {
```

Comment:
where does the `R` in the name comes from?

Reply:
R stands for 'recursive'. I'll type it out. (or, maybe another name like `scalarLayoutsInternal` is better. That seems to be more common in JDK code at least)

Reply:
I was about to suggest `internal` - but then I wasn't sure if it was derived from an ABI specific term.